File: euler-2.js
function fibonacci(first, second, numbers, max) {
  var generatedNumbers = numbers;
  if (first + second <= max) {
    var sum = first + second;
    numbers.push(sum);
    return fibonacci(second, sum, generatedNumbers, max)
  } else {
    return generatedNumbers
  }
}

var evens = fibonacci(1, 1, [1,1], 4000000).filter(function(a) { return a % 2 == 0 });

console.log(evens.reduce(function(a,b) { return a + b}));

File: rb/euler_2.rb
$fib = [1,1]
Max = 4000000

def generate_fibonacci(fib)
  nextFib = caclulate_next_fibb(fib[-1], fib[-2])
  if nextFib >= Max
    return fib
  else
    fib.push(nextFib)
    generate_fibonacci(fib)
  end
  fib
end

def caclulate_next_fibb(first, second)
  first + second
end

def sum_evens(fibs)
  fibs.select(&:even?).reduce(:+)
end

puts sum_evens(generate_fibonacci($fib))
